ex-3/chatclient/lib.c

```diff
@@ -0,0 +1,39 @@
+#ifndef __COMMON_LIB__
+
+#define __COMMON_LIB__
+#define BUFFER_LENGTH 1000
+
+char* GetLineFromSocket(int socket){
+    char c;
+    char* sp = (char*)malloc(BUFFER_LENGTH);
+    int i = 0;
+
+    while (recv(socket, &c, sizeof c, 0) == 1) {
+        if (c == '\n') {
+            sp[i] = 0;
+            return sp;
+        }
+        else {
+            sp[i++] = c;
+        }
+    }
+
+    return 0;
```

Comment:
return '\0' or 0 ?

Reply:
return 0, char* address, not char.

Reply:
OK. Then let's use NULL or nullptr for functions which return a pointer.